routes/main.js

```diff
@@ -12,38 +13,65 @@ exports.main = {
   config: {
     validate: {
       query: {
-        url: Joi.string().required(),
-        viewport: Joi.string().optional()
+        url: Joi.string().uri().required(),
+        viewport: Joi.string().optional(),
+        pixelRatio: Joi.number().integer().min(1).optional(),
+        wsEndpoint: Joi.string().optional()
       }
     }
   },
-  handler(request, reply) {
+  async handler(request, reply) {
     const server = request.server;
     const url = request.query.url;
     const randFldr = rando(18);
     const dir = `./screenshots/${randFldr}`;
+    const outputFile = `${dir}/screenshot.png`;
+    
+    try {
+      let browser;
 
-    const command = [];
+      await fs.ensureDir(dir);
+      
+      if (request.query.wsEndpoint) {
+        console.log(`Connecting to browser websocket at "${request.query.wsEndpoint}".`);
+        browser = await puppeteer.connect({
+          browserWSEndpoint: request.query.wsEndpoint
+        });
+      } else {
+        browser = await puppeteer.launch();
```

Comment:
Should add and ENV Var as a default so we don't have to pass the `wsEndpoint` to the app every time.